nixos/modules/system/boot/stage-1-init.sh

```diff
@@ -614,10 +614,9 @@ echo /sbin/modprobe > /proc/sys/kernel/modprobe
 
 
 # Start stage 2.  `switch_root' deletes all files in the ramfs on the
-# current root.  Note that $stage2Init might be an absolute symlink,
-# in which case "-e" won't work because we're not in the chroot yet.
-if [ ! -e "$targetRoot/$stage2Init" ] && [ ! -L "$targetRoot/$stage2Init" ] ; then
-    echo "stage 2 init script ($targetRoot/$stage2Init) not found"
+# current root.  The path has to be valid in the chroot not outside.
+if ! chroot "$targetRoot" [ -e "$stage2Init" ] ; then
+    echo "stage 2 init script ($stage2Init under $tagetRoot) not found"
```

Comment:
```suggestion
    echo "stage 2 init script ($stage2Init under $targetRoot) not found"
```

Reply:
Is it possible to use shellcheck on this script?

Reply:
Good catch! I just finished running `shellcheck` and it only reports what you already noticed for the changed lines
```
In stage-1-init.sh line 619:
    echo "stage 2 init script ($stage2Init under $tagetRoot) not found"
                                                 ^--------^ SC2154: tagetRoot is referenced but not assigned (did you mean 'targetRoot'?).
```

I've force pushed a correct version. Thanks! :+1: